convert2rhel/unit_tests/pkghandler_test.py

```diff
@@ -1453,3 +1450,176 @@ def test_fix_default_kernel_with_no_incorrect_kernel(self):
 kernel.x86_64    4.7.4-200.fc24   @updates
 Available Packages
 kernel.x86_64    4.7.4-200.fc24   @updates"""
+
+with open(
+    os.path.join(os.path.dirname(__file__), "data/pkghandler_yum_distro_sync_output_expect_deps_for_3_pkgs_found.txt")
+) as f:
+    YUM_DISTRO_SYNC_OUTPUT = f.read()
+
+
+@pytest.mark.parametrize(
+    "yum_output, expected",
+    (
+        (
+            "",
+            {
+                "protected": set(),
+                "errors": set(),
+                "multilib": set(),
+                "required": set(),
+                "mismatches": set(),
+            },
+        ),
+        (
+            YUM_PROTECTED_ERROR,
+            {
+                "protected": set(("systemd", "yum")),
+                "errors": set(),
+                "multilib": set(),
+                "required": set(),
+                "mismatches": set(),
+            },
+        ),
+        (
+            YUM_MULTILIB_ERROR,
+            {
+                "protected": set(),
+                "errors": set(),
+                "multilib": set(("libstdc++", "openldap", "p11-kit")),
+                "required": set(),
+                "mismatches": set(),
+            },
+        ),
+        (
+            YUM_MISMATCHED_PKGS_ERROR,
+            {
+                "protected": set(),
+                "errors": set(),
+                "multilib": set(),
+                "required": set(),
+                "mismatches": set(("python39-psycopg2-debug",)),
+            },
+        ),
+        # These currently do not pass because the Requires handling
+        # misparses a Requires in the test data (It turns
+        # python2_hawkey into python2).  Testing it in its own function
+        # for now.
+        # (
+        #     YUM_REQUIRES_ERROR,
+        #     {
+        #         "protected": set(),
+        #         "errors": set(("gcc-c++", "libreport-anaconda", "abrt-cli", "mod_ldap", "redhat-lsb-trialuse")),
+        #         "multilib": set(),
+        #         "required": set(("gcc", "libreport-plugin-rhtsupport", "python2-hawkey", "redhat-lsb-core")),
```

Comment:
<nod> It would depend on what the actual packages are.  You would be correct if there are two separate packages, python2-hawkey and python2_hawkey.  In my mind I was thinking that the package is python2-hawkey and it has a virtual provide/capability named python2_hawkey.  (Like this in the spec file:

```
Name: python2-hawkey
# [..]
Provides: python2_hawkey = %{version}-%{release}
```
This situation would arise when a package has been renamed or the packager thinks people would sometimes use the alternate name for some other reason.

Reply:
I was referring to the test yum output https://github.com/abadger/convert2rhel/blob/d24e424da77636af37e38881d797f5326da8f383/convert2rhel/unit_tests/pkghandler_test.py#L1381 where we have `Requires: python2_hawkey >= 0.7.0`.

Such `Provides:` in reality (in hawkey spec file) does not exist. We came up with the `Requires: python2_hawkey` for the sake of testing. However we failed to write the test properly as you revealed. We should have added 
`self.assertIn("python2_hawkey", error_pkgs["required"])` to the test in
https://github.com/oamg/convert2rhel/pull/325/files#diff-c3ce6b0e1e7ddae0678f9eb7c71f9c5f8aecbfd5b7db1deb7d8947ccf1492a08R314.